extensions/amp-position-observer/amp-position-observer.md

```diff
@@ -0,0 +1,220 @@
+  <script type="application/json">
+    {
+    "duration": "3s",
+    "fill": "both",
+    "direction": "alternate",
+    "animations": [
+      {
+        "selector": "#clock-scene .clock-hand",
+        "keyframes": [
+          { "transform": "rotate(-180deg)" },
+          { "transform": "rotate(0deg)" }
+        ]
+      }
+    ]
+  }
+  </script>
+</amp-animation>
+
+<!-- The clock container -->
+<div id="clock-scene">
+  <amp-img layout="responsive" width=2 height=1.5 src="./img/clock.jpg">
+    <div class="clock-hand"></div>
+  </amp-img>
+</div>
+
+<!--
+  Use amp-position-observer to tie the movement of the clock scene within
+  the viewport to the timeline of the animation
+-->
+
+<amp-position-observer
+  target-selector="#clock-scene"
+  intersection-ratios="1"
+  on="scroll:clockAnim.seekTo(percent=event.percent)"
+  layout="nodisplay">
+</amp-position-observer>
+```
+
+### Animation scenes that start/pause based on visibility in the viewport
+`amp-animation` also exposes `start` and `pause` actions that can be tied to the
+`enter` and `exit` events of `amp-position-observer` to control when animation
+starts/pauses based on visibility.
+
+`amp-position-observer` exposes various visibility configurations, such as
+`intersection-ratios` and `exclusion-margins` (similar to [IntersectionObserver](https://developer.mozilla.org/en-US/docs/Web/API/Intersection_Observer_API)) that
+can be used to fine-tune when the target is considered visible.
+
+## Example
+Consider the same clock animation, but this time the hand animates with time, except
+we like the animation to start when clock is at least 50% visible and pause as soon
+as clock becomes less than 50% visible.
+
+![visibility demo](https://user-images.githubusercontent.com/2099009/29105727-a7d9a80a-7c84-11e7-8d4a-794f38ea5a5c.gif)
+
+
+```html
+<!-- An animation that rotates a clock hand 180 degrees. -->
+<!--
+   Note that we are NOT setting `trigger=visibility`
+   since we will manually trigger the animation
+-->
+<!--
+   Also note that this is the same animation as the scroll-bound version above
+   the animation is the same, just the triggering mechanism with
+   `amp-position-observer` is different!
+-->
+<amp-animation id="clockAnim" layout="nodisplay">
+  <script type="application/json">
+    {
+    "duration": "3s",
+    "fill": "both",
+    "direction": "alternate",
+    "animations": [
+      {
+        "selector": "#clock-scene .clock-hand",
+        "keyframes": [
+          { "transform": "rotate(-180deg)" },
+          { "transform": "rotate(0deg)" }
+        ]
+      }
+    ]
+  }
+  </script>
+</amp-animation>
+
+<!-- The clock container -->
+<div id="clock-scene">
+  <amp-img layout="responsive" width=2 height=1.5 src="./img/clock.jpg">
+    <div class="clock-hand"></div>
+  </amp-img>
+</div>
+
+<!--
+  Use amp-position-observer to tie the start/pause of the animation with
+  the visibility of the scene.
+-->
+
+<amp-position-observer
```

Comment:
Optional, but I'd put observer inside the scene and drop selector here. I think that'd be a more canonical use. But feel free to disregard.